framework/resource/resource.go

```diff
@@ -149,6 +152,42 @@ func (r *Resource) Destroy(args ...interface{}) error {
 	return result.Err()
 }
 
+// Status returns a copy of this resources' status, or nil if no status exists.
+func (r *Resource) Status() *pb.StatusReport_Resource {
+	if r.status == nil {
+		return nil
+	}
+	// return a copy to avoid mutating status
+	return &pb.StatusReport_Resource{
+		Name:          r.status.Name,
+		Health:        r.status.Health,
+		HealthMessage: r.status.HealthMessage,
+	}
+}
+
+// GetStatus creates an individual status report for this resource.
+//
+// After GetStatus has been called, any status can be accessed via the Status
+// function.
+func (r *Resource) GetStatus(args ...interface{}) error {
```

Comment:
Same feedback as manager: feels weird to have two functions, why not just return the status report when calling this?

Reply:
After we discussed this internally, I've renamed `GetStatus()` to just `status()` and documented that it's only used for testing. Changes made in 5632677